src/Build/Evaluation/IntrinsicFunctions.cs

```diff
@@ -177,29 +179,36 @@ internal static object GetRegistryValue(string keyName, string valueName, object
             return Registry.GetValue(keyName, valueName, defaultValue);
         }
 
+        internal static object GetRegistryValueFromView(string keyName, string valueName, object defaultValue, params object[] views)
+        {
+            if (views == null || views.Length == 0)
+            {
+                views = DefaultRegistryViews;
+            }
+
+            return GetRegistryValueFromView(keyName, valueName, defaultValue, new ArraySegment<object>(views));
+        }
+
+
         /// <summary>
         /// Get the value of the registry key from one of the RegistryView's specified
         /// </summary>
-        internal static object GetRegistryValueFromView(string keyName, string valueName, object defaultValue, params object[] views)
+        internal static object GetRegistryValueFromView(string keyName, string valueName, object defaultValue, ArraySegment<object> views)
```

Comment:
What's the advantage of `ArraySegment`? I don't think I've used it before.

Reply:
It's like Span and Memory, but from the distant net46 past.